server/src/main/java/org/elasticsearch/index/mapper/RangeFieldMapper.java

```diff
@@ -284,7 +286,7 @@ public Query termQuery(Object value, QueryShardContext context) {
 
         @Override
         public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower, boolean includeUpper,
-                                ShapeRelation relation, DateTimeZone timeZone, DateMathParser parser, QueryShardContext context) {
+                                ShapeRelation relation, DateTimeZone timeZone, JodaDateMathParser parser, QueryShardContext context) {
```

Comment:
I may be misinterpreting this: Shouldnt this be the DateMathParser as argument in the fieldmappers, so that the underlying implementation can be replaced?

Reply:
Yes, this was due to simple renaming with IntelliJ. I've converted these signatures back to the interface in d996081.